spec/unit/pops/validator/validator_spec.rb

```diff
@@ -153,6 +153,76 @@ def validate(factory)
     end
   end
 
+  context 'with --tasks set' do
+    before(:each) { Puppet[:tasks] = true }
+
+    it 'produces an error for application' do
+      acceptor = validate(parse('application test {}'))
+      expect(acceptor.error_count).to eql(1)
+      expect(acceptor).to have_issue(Puppet::Pops::Issues::UNSUPPORTED_TASKS_EXPRESSION)
+    end
+
+    it 'produces an error for capability mapping' do
+      acceptor = validate(parse('Foo produces Sql {}'))
+      expect(acceptor.error_count).to eql(1)
+      expect(acceptor).to have_issue(Puppet::Pops::Issues::UNSUPPORTED_TASKS_EXPRESSION)
+    end
+
+    it 'produces an error for collect expressions' do
+      acceptor = validate(parse("User <| title == 'admin' |>"))
+      expect(acceptor.error_count).to eql(1)
+      expect(acceptor).to have_issue(Puppet::Pops::Issues::UNSUPPORTED_TASKS_EXPRESSION)
+    end
```

Comment:
also test double UFO `<<| |>>` expression to be sure both are covered. Don't think also testing both with overridden attributes makes a difference.